hs/src/Reach/Compiler.hs

```diff
@@ -69,15 +69,15 @@ compile env (CompilerOpts {..}) = do
         ll <- linearize showp dl
         ol <- optimize ll
         showp "ol" ol
-        unless (not co_sim) $ do
-          startServer ol
         let vo_out = woutnMay
         let vo_mvcs = doIf connectors dlo_verifyPerConnector
         let vo_timeout = co_verifyTimeout
         let vo_dir = dirDotReach'
         verify (VerifyOpts {..}) ol >>= maybeDie
         el <- erase_logic ol
         showp "el" el
+        unless (not co_sim) $ do
+          startServer el
```

Comment:
One danger of this is that I have a future dream of integrating the simulator logic with the SMT solver so you can do something like "return a symbolic value" rather than 42 for a given interact and then get a SMT context. I don't think you should change anything, but it's something I think about